src/test/run-pass/segfault-no-out-of-stack.rs

```diff
@@ -15,10 +15,59 @@ extern crate libc;
 use std::process::{Command, ExitStatus};
 use std::env;
 
+#[cfg(unix)]
+unsafe fn segfault() {
+    // Map and unmap a page of memory to guarantee that we have a pointer that
+    // will produce a segfault.
+    let pagesize = libc::sysconf(libc::_SC_PAGESIZE);
+    if pagesize == -1 {
+        panic!("Couldn't get pagesize");
+    }
+
+    let mut buf = libc::mmap(std::ptr::null_mut(),
+                             pagesize as libc::size_t,
+                             libc::PROT_WRITE | libc::PROT_READ,
+                             libc::MAP_ANON | libc::MAP_PRIVATE,
+                             -1,
+                             0);
+
+    if buf == libc::MAP_FAILED {
+        panic!("mmap failed");
+    }
+
+    let res = libc::munmap(buf, pagesize as libc::size_t);
+
+    *(buf as *mut libc::c_char) = 1;  // trigger the actual segfault
+}
+
+#[cfg(not(unix))]
+unsafe fn segfault() {
+    // This isn't actually guaranteed to produce a segfault; it appears that LLVM
+    // sometimes just compiles this down to an illegal instruction.  Will need to
+    // provide an implementation of something like the above in order for this test
+    // to be valid on non-Unix platforms.
+    *(0 as *mut isize) = 1;
```

Comment:
An easier way to write this is probably also to do something like add a helper function to `rust_test_helpers.c` which just returns the null pointer and then dereference that. LLVM can't prove the pointer coming from C is the null pointer so it won't be able to do this optimization.